builder/optimizations.cc

```diff
@@ -55,6 +55,24 @@ bool is_copy(var src, expr src_x, int src_d, var dst, var dst_x, int dst_d, expr
     src_dim.fold_factor = dim::unfolded;
     return true;
   } else {
+    expr fold_factor = buffer_fold_factor(src, src_d);
+    bool is_mod = false;
+
+    // Try to parse src_x = dst_x % fold_factor
+    if (const class mod* m = src_x.as<class mod>()) {
+      const bool is_unfolded =
+          !src_dim.fold_factor.defined() ||
+          src_dim.fold_factor.same_as(dim::unfolded);
+      if (depends_on(m->a, dst_x).any() && !depends_on(m->b, dst_x).any() &&
```

Comment:
Nit: put the is_unfolded first in the condition (short circuit the relatively expensive `depends_on` checks)